src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -156,6 +157,24 @@ public abstract class Index : IDocumentTombstoneAware, IDisposable
             Suggestion = "Please verify this index definition and consider a re-design of your entities or index for better indexing performance"
         };
 
+        public const string DynamicIndex = "dynamic";
+
+        public const string DynamicIndexPrefix = "dynamic/";
+
+        public static bool IsDynamicIndex(string indexName)
+        {
+            if (indexName == null || indexName.Length < DynamicIndex.Length)
```

Comment:
I took that code from somewhere else and just refactored it in a 'more proper?' place. The underlying issues of using only ```.StartWidth``` is that you cant use ```dynamic``` (disallowed for some reason) and you dont want to match ```dynamicIndexWhatever``` as dinamic.